packages/client/test/client.spec.ts

```diff
@@ -244,3 +248,102 @@ describe('test hide property', () => {
     ).to.be.equal(true);
   });
 });
+
+type Listener = (m: MessageEvent) => unknown;
+
+class MessageEvent {
+  data: { [key: string]: string } = {};
+  source: Window = window;
+
+  constructor(
+    type: string,
+    eventInitDict: { [key: string]: { [key: string]: string } },
+  ) {
+    if (type === 'message') {
+      this.data = eventInitDict.data;
+    }
+  }
+}
+
+function mockWindow(listenerQueue: Listener[]) {
+  return {
+    addEventListener: (type: string, listener: Listener, option: boolean) => {
+      if (type === 'message') {
+        listenerQueue.push(listener);
+        return option;
+      }
+    },
+    postMessage: (message: { [key: string]: string }, targetOrigin: string) => {
+      if (targetOrigin !== undefined) {
+        const event = new MessageEvent('message', { data: message });
+        listenerQueue.forEach((listener) => listener(event));
+      }
+    },
+  };
+}
```

Comment:
Declaration for mocking. Because node.js environment doesn't have `window` object.